tests/rados/test_osd_args.py

```diff
@@ -0,0 +1,77 @@
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-10417
+    #BZ-1331770
+    This test is to verify negative scenarios and inputs for
+    'ceph osd' commands as listed below:
+    - ceph osd reweight-by-pg
+    - ceph osd reweight-by-utilization
+    - ceph osd test-reweight-by-pg
+    - ceph osd test-reweight-by-utilization
+    1. Create cluster with default configuration
+    2. Execute the commands with invalid inputs
+    3. Ensure that execution with invalid input fails
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    cmd_list = config["cmd_list"]
+    base_cmd = "ceph osd"
+    oload_invalid = [100, -10, 0, "abcd"]
+    max_invalid = [-10, 0, "abcd"]
+
+    def execute_cmd_verify() -> bool:
+        """
+        Executes the input command on installer node,
+        inside cephadm shell and verifies negative scenarios.
+        :param cmd: (string) command to run
+        :param value: invalid value for the cmd
+        :return: True -> pass | False -> fail
+        """
+        log.info(
+            f"Executing '{exec_cmd}' on {cephadm.installer.node.hostname} | Expected to fail"
+        )
+        out, err = cephadm.shell(args=[exec_cmd], check_status=False)
+        if type(value) == str and (
+            f"Invalid command: {value} doesn't represent a float" in err
+            or f"Invalid command: {value} doesn't represent an int"
+        ):
+            return True
+        else:
+            if "oload" in exec_cmd and (
+                "Error EINVAL: You must give a percentage higher than 100." in err
+                or "FAILED reweight-by-" in err
+            ):
+                return True
+            if f"{value} must be positive" in err:
+                return True
+        return False
+
+    try:
+        for _cmd in cmd_list:
+            for value in oload_invalid:
```

Comment:
Please add Debug statements for the actions that you would start and test, so that those details are captured?

Reply:
As discussed offline, adding command specific statements would require multiple conditional checks. This can be taken up later when valid scenarios are added. I have taken a note of this.